src/main/java/org/broadinstitute/consent/http/service/DacService.java

```diff
@@ -180,6 +180,15 @@ public void updateDac(String name, String description, String email, Integer dac
   }
 
   public void deleteDac(Integer dacId) {
+    Dac fullDac = dacDAO.findById(dacId);
+    List<DataAccessAgreement> daas = daaDAO.findAll();
+    List<Dac> dacs = dacDAO.findAll();
+    daas.stream()
+        .filter(daa -> daa.getInitialDacId().equals(dacId) && ! daaService.isBroadDAA(daa.getDaaId(), daas, dacs))
+        .forEach(daa -> daaService.deleteDaa(daa.getDaaId()));
+    if (daaService.isBroadDAA(fullDac.getAssociatedDaa().getDaaId(), daas, dacs)) {
+      daaService.removeDacFromDaa(dacId, fullDac.getAssociatedDaa().getDaaId());
+    }
```

Comment:
This approach doesn't group all of the deletes into a logical batch, but runs them individually. Instead, we should be batching them up in a single transaction. For example, have a look at [NihServiceDAO](https://github.com/DataBiosphere/consent/blob/e03a530e474446dbc848797cde82116128280346/src/main/java/org/broadinstitute/consent/http/service/dao/NihServiceDAO.java#L26) for an approach that keeps all of the updates in a single transaction. In the case of `DacService`, we don't have a service dao, so this is a good chance to add one.

Another option is to wrap everything up in a single statement similarly to what we do with DAA deletes in [DaaDAO](https://github.com/DataBiosphere/consent/blob/e03a530e474446dbc848797cde82116128280346/src/main/java/org/broadinstitute/consent/http/db/DaaDAO.java#L196), but that might be a little less maintainable than the service DAO approach.